app/javascript/admin/controllers/config_controller.js

```diff
@@ -93,6 +93,40 @@ export default class ConfigController extends Controller {
     }
   }
 
+  displaySnackbar(message) {
+    return document.dispatchEvent(
+      new CustomEvent('snackbar:add', { detail: { message } }),
+    );
+  }
+
+  async activateLightConfirmation(event) {
+    event.preventDefault();
+    try {
+      const data = new FormData(event.target);
+      const response = await fetch(event.target.action, {
+        method: 'POST',
+        headers: {
+          Accept: 'application/json',
+          'X-CSRF-Token': document.querySelector("meta[name='csrf-token']")
+            ?.content,
+        },
+        body: data,
+        credentials: 'same-origin',
+      });
+
+      const outcome = await response.json();
+
+      if (response.ok) {
+        this.displaySnackbar(outcome.message);
+      } else {
+        this.displaySnackbar(outcome.error);
+      }
+    } catch (err) {
+      this.displaySnackbar(err);
```

Comment:
I tried to test out the error path by blocking one of the network requests in dev tools - I can see that we're hitting this code with "TypeError: Failed to fetch", but the snackbar shows up as empty 🙃  Not sure what's up here -

https://user-images.githubusercontent.com/20773163/130247152-1e51c6b9-f466-48b0-b03f-8dd5671683f7.mp4

I'm also wondering from the UX side of things, should we be drawing a user's attention to the input field when something doesn't update? Or change the input value back to the old one in the UI to reflect that the request failed? I think it's quite easy to lose track of what failed and where, especially now it's async - if I hit submit and scroll away I have to peep this error message and remember what on earth I was trying to update 😅

Reply:
Hmmmm I'm not sure what's up with the `err` not showing in the snackbar 🤔 

I think your UX observations are spot on; I'm not sure how I would "revert" the user's entries back to the old ones without a refresh though...

Reply:
Yeah it's true, it would add quite a layer of complexity to try to revert the entries (and maybe not worth it anyway - would a user notice it'd changed back?)

I suggested it above but maybe an `error` class could be added to highlight it in red with an `!` icon or something 🤔  Can definitely be a future enhancement though

Reply:
Ok I have made a note to address highlighting failed updates. I also fixed the `err` not showing in the snackbar - I needed to call `err.message` 😅